src/libraries/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -973,7 +973,7 @@ public override async Task WaitForCancellationAsync(bool ignoreIncomingData = tr
                 {
                     Assert.Equal(FrameType.RstStream, frame.Type);
                 }
-            } while (frame.Type != FrameType.RstStream);
+            } while (frame.Type != FrameType.RstStream || (frame is RstStreamFrame rstStreamFrame && rstStreamFrame.ErrorCode != (int)ProtocolErrors.CANCEL));
```

Comment:
We should not receive any RstStream frame other than one containing the CANCEL.

As written, this will just ignore a RstStream that doesn't contain CANCEL.

Reply:
So we should rather just assert that the error code is CANCEL, is that what you mean @geoffkizer?

Reply:
Yes.